.circleci/config.yml

```diff
@@ -1,79 +1,121 @@
+      $CIRCLE_TEST_REPORTS/server
+
+
 version: 2
 jobs:
-  build:
-    parallelism: 6
+  #
+  # Shared setup required for all jobs
+  #
+  setup:
+    docker: *docker-image
     working_directory: ~/wp-calypso
-    docker:
-      - image: circleci/node:10.10.0-browsers
-        environment:
-          CIRCLE_ARTIFACTS: /tmp/artifacts
-          CIRCLE_TEST_REPORTS: /tmp/test_results
-
     steps:
-      - restore_cache:
-          name: Restore git cache
-          key: v1-20180524-git
-
+      - restore_cache: *restore-git-cache
       - checkout
-
-    # Update origin/master so we can diff against it later
+      # Update origin/master so we can diff against it later
       - run:
           name: Update master branch
           command: git fetch --force origin master
+      - save_cache: *save-git-cache
 
-      - save_cache:
-          name: Save git cache
-          key: v1-20180524-git
-          paths:
-            - ".git"
-
-      - run:
-          name: Create Directories for Results and Artifacts
-          command: |
-            mkdir -p $CIRCLE_ARTIFACTS
-            mkdir -p $CIRCLE_TEST_REPORTS
-
-      - restore_cache:
-          name: "Restoring npm cache"
-          keys:
-             - v1-npmcache-{{ checksum ".nvmrc" }}-{{ checksum "npm-shrinkwrap.json" }}
-             - v1-npmcache-{{ checksum ".nvmrc" }}
-
+      - restore_cache: *restore-npm-cache
       - run: npm ci
+      - save_cache: *save-npm-cache
+      - persist_to_workspace:
+          root: .
+          paths: .
+
+  #
+  # Build translations
+  #
+  translate:
+    docker: *docker-image
+    working_directory: ~/wp-calypso
+    environment: *shared-environment
 
-      - save_cache:
-          name: "Saving npm cache"
-          key: v1-npmcache-{{ checksum ".nvmrc" }}-{{ checksum "npm-shrinkwrap.json" }}
-          paths:
-            - ~/.npm
+    steps:
+      - attach_workspace:
+          at: .
 
-      - run:
-          name: Build Server
-          environment:
-              NODE_ENV: test
-          command: npm run build-server
+      - run: *setup-results-and-artifacts
 
       - run:
           name: Build calypso-strings.pot
           command: |
-            if [ "$CIRCLE_NODE_INDEX" == "0" ]; then
-              npm run translate; mkdir -p $CIRCLE_ARTIFACTS/translate
-              mv calypso-strings.pot $CIRCLE_ARTIFACTS/translate
-            fi
+            npm run translate; mkdir -p $CIRCLE_ARTIFACTS/translate
+            mv calypso-strings.pot $CIRCLE_ARTIFACTS/translate
 
       - run:
           name: Build New Strings .pot
           command: |
-            if [ "$CIRCLE_NODE_INDEX" == "0" ]; then
-              git clone https://github.com/Automattic/gp-localci-client.git
-              bash gp-localci-client/generate-new-strings-pot.sh $CIRCLE_BRANCH $CIRCLE_SHA1 $CIRCLE_ARTIFACTS/translate
-              rm -rf gp-localci-client
-            fi
+            git clone https://github.com/Automattic/gp-localci-client.git
```

Comment:
would it be helpful here to perform a shallow clone? I don't think we need the history do we?

```bash
git clone --depth=1 https://github.com/Automattic/gp-localci-client.git
```

Would we have any benefit to caching this and doing a fetch instead?

Reply:
> would it be helpful here to perform a shallow clone? I don't think we need the history do we?

😁 I agree (#26226):

https://github.com/Automattic/wp-calypso/blob/f851182623c77178af75b51dda621aeed6ec84b8/.circleci/config.yml#L65

> Would we have any benefit to caching this and doing a fetch instead?

The repo we're checking out is small enough that the entire step often completes with a duration of `0:00`. It's not really worth optimizing:

![zero](https://user-images.githubusercontent.com/841763/45803217-4c4ebb80-bcb9-11e8-8d55-0f9c4c0e5e43.png)

I wish the repo didn't even exist. It's just a bash script that's shared between a couple of projects.